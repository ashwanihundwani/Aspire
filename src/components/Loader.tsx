
import React from 'react'
import { View, ActivityIndicator, StyleSheet } from 'react-native'

interface LoaderProps {
    isLoading?: boolean
    style?: any
}
const Loader: React.FC<LoaderProps> = (props: LoaderProps) => {
    if (!props.isLoading) {
        return null
    }
    return (
        <View style={props.style ? props.style : styles.container}>
            <ActivityIndicator size={'large'} color={'white'} animating />
        </View>
    )
}

const styles = StyleSheet.create({
    container: {
        position: 'absolute',
        backgroundColor: 'rgba(11, 20, 44, 0.55)',
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
        alignItems: 'center',
        justifyContent: 'center',
    },
})

export default Loader
