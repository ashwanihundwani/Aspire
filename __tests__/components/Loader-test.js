/**
 * @format
 */

import 'react-native';
import React from 'react';
import Loader from '../../src/components/Loader';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer
      .create(<Loader isLoading={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
});