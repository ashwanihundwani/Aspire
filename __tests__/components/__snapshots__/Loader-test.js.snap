// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(11, 20, 44, 0.55)",
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="white"
    size="large"
  />
</View>
`;
